docs/src/proving_systems/starks/protocol.md

```diff
@@ -1,64 +1,101 @@
+3. Let $r_\text{min}$ and $r_\text{max}$ be respectively the minimum and maximum values of the entries of the submatrix $M_\text{offsets}$ defined by the columns of the group `offsets`. Let $v$ be the vector of all the values between $r_\text{min}$ and $r_\text{max}$ that are not in $M_\text{offsets}$. If the length of $v$ is not a multiple of three, extend it to the nearest multiple of three using one arbitrary value of $v$.
+
+4. Let $L_v$ be the length of $v$. Add $L_v$ rows repeating the last row of $M$ in all the columns except in the group `offsets`. In that section place the values of $v$. The result is a matrix $M$ in $\mathbb{F}^{(L + L' + L_v) \times 33}$
 
-7. Pad $M$ with copies of its last row until it has a power of two number of rows. As a result we obtain a matrix $M_{\text{RAP1}}\in\mathbb{F}^{2^n\times 33}$.
+5. Pad $M$ with copies of its last row until it has a power of two number of rows. As a result we obtain a matrix $M_{\text{RAP1}}\in\mathbb{F}^{2^n\times 33}$.
 ##### Second round of RAP:
 
 The verifier sends challenges $\alpha, z \in \mathbb{F}$ (or the prover samples them from the transcript). Additional columns are added to incorporate the memory constraints. To define them the prover follows these steps:
-1. Stack the rows of the submatrix of $M$ defined by the columns `pc, dst_addr, op0_addr, op1_addr` into a vector `a` of length $4(L+L')$ (this means that the first entries of `a` are `pc[0], dst_addr[0], op0_addr[0], op1_addr[0], pc[1], dst_addr[1],...`).
-2. Stack the the rows of the submatrix defined by the columns `inst, dst, op0, op1` into a vector `v` of length $4(L+L')$.
-3. Define $M_{\text{Mem}}\in\mathbb{F}^{(L+L')\times 2}$ to be the matrix with columns `a`, `v`.
-3. Define $M_{\text{MemRepl}}\in\mathbb{F}^{(L+L')\times 2}$ to be the matrix that's equal to $M_{\text{Mem}}$ in the first $L - L_{\text{pub}}$ rows, and its last $L_{\text{pub}}$ entries are the addresses and values of the actual public memory.
-5. Sort $M_{\text{MemRepl}}$ by the first column in increasing order. The result is a matrix $M_{\text{MemReplSorted}}$ of size $(L+L')\times 2$. Denote its columns by `a'` and `b'`.
-6. Compute the vector $p$ of size $4(L+L')$ with entries 
-$$ p_i = \prod_{j=0}^i\frac{z - (\alpha a_i' + v_i')}{z - (\alpha a_i + v_i)}$$
-5. Reshape the matrix $M_{\text{MemReplSorted}}$ into a $(L+L')\times8$ in row-major. Reshape the vector $p$ into a $(L+L')\times4$ matrix in row-major.
-6. Concatenate these 12 rows. The result is a matrix $M'$ of size $(L+L') \times 12$.
-7. Pad $M'$ with copies of its last row until it has a power of two number of rows. As a result we obtain a matrix $M_{\text{RAP2}}\in\mathbb{F}^{2^n\times 12}$.
+1. Stack the rows of the submatrix of $M_{\text{RAP1}}$ defined by the columns `pc, dst_addr, op0_addr, op1_addr` into a vector `a` of length $2^{n+2}$ (this means that the first entries of `a` are `pc[0], dst_addr[0], op0_addr[0], op1_addr[0], pc[1], dst_addr[1],...`).
```

Comment:
Tip: Markdown supports numbered lists to be defined only as `1.` for all items. It can then fix the numbers itself. This is useful in case in the future you need to add some item in the middle as it saves you from updating all one by one.

Reply:
Nice! Didn't know that